src/frontend/src/utils/pretty_serde.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! @kwannoel:
+//! This module implements Serde for the Pretty struct. Why not implement it directly on our plan nodes?
+//! That's because Pretty already summarizes the fields that are important to us.
+//! You can see that when `explain()` is called, we directly return the `Pretty` struct.
+//! The _proper way_ to do this would be to create a new data structure that plan nodes get converted into,
```

Comment:
You mean:

```
`Plan` -> `PlanForExplain` -> `Pretty` -> `String`
                          |-> `Json`
```

Interesting

Reply:
I agree `Pretty` is not targeted as the real "PlanForExplain" IR, but good enough for now 👍